ext/spl/spl_iterators.c

```diff
@@ -1456,6 +1456,7 @@ static spl_dual_it_object* spl_dual_it_construct(INTERNAL_FUNCTION_PARAMETERS, z
 						zval_ptr_dtor(&retval);
 						return NULL;
 					}
+					/* TODO Change to normal behaviour */
 					if (Z_TYPE(retval) != IS_OBJECT || !instanceof_function(Z_OBJCE(retval), zend_ce_traversable)) {
 						zend_throw_exception_ex(spl_ce_LogicException, 0, "%s::getIterator() must return an object that implements Traversable", ZSTR_VAL(ce->name));
 						return NULL;
```

Comment:
I this a type error or some other kind of error?

Reply:
It seems to be a type error for me.

Reply:
Seems like, will change this and force push again.